### ISSUE TITLE
on_change event for cells

### ISSUE THREAD

**Issue by emirror-de** [Synphonyte/leptos-struct-table#3]
Hi there!

I would like to render a table where the user is able to edit the content of the cells. I already noticed the `on_click` event on the `RowRenderer` which is basically what I am looking for, but on change in the cell renderer.

Something like:

```rust
fn CustomCellEditRenderer(...) -> impl IntoView {
    view!{
        cx,
        <td>
            <select on:change=move |_| on_change(/* parameters that contain information like row key, column and updated value */)>
                <option value="1">Something</option>
                <option value="2">Else</option>
            </select>
        </td>
    }
}
```

Is there a simple way to do this?

**Comment by maccesch** [comment]
Editing cells is definitely on the roadmap but right now it’s not implemented yet. We’re open to a PR though

**Comment by emirror-de** [comment]
Sounds good, I would be happy to give it a try. Can you give me a short guideline what needs to be changed or where to look at first, so I have a starting point?

**Comment by emirror-de** [comment]
Thanks for your explanations, they have been very helpful for me to get started!

I prepared a PR for each of the repositories:
https://github.com/Synphonyte/leptos-struct-table/pull/4
https://github.com/Synphonyte/leptos-struct-table-macro/pull/1

The first one contains explanations about the changes and my thoughts on it.

**Comment by emirror-de** [comment]
In addition to the updates in #4 I already have some other things added like support for `DateTime<Utc>` and double click event on rows. You can have a look [here](https://github.com/emirror-de/leptos-struct-table/tree/feature/opus-portal). For the macro crate the same branch exists.
Still the `server_function` example is not updated yet, but if you are happy with the additions I did we can replace #4 with the `feature/opus-rs` branch.

**Comment by maccesch** [comment]
I like your changes in the `feature/opus-rs` branch. Let's go back to the `TableDataProvider` as it was with the synchronous `set_row` method and I'll merge everything. We can figure out the async later.

**Comment by maccesch** [comment]
Implemented in 34db429
